karate-core/src/main/java/com/intuit/karate/http/ApacheHttpClient.java

```diff
@@ -285,35 +337,36 @@ public Response invoke(HttpRequest request) {
         }
         if (request.getHeaders() != null) {
             request.getHeaders().forEach((k, vals) -> vals.forEach(v -> requestBuilder.addHeader(k, v)));
-        }        
-        CloseableHttpResponse httpResponse;
-        byte[] bytes;
-        try (CloseableHttpClient client = clientBuilder.build()) {
-            httpResponse = client.execute(requestBuilder.build());
-            HttpEntity responseEntity = httpResponse.getEntity();
-            if (responseEntity == null || responseEntity.getContent() == null) {
-                bytes = Constants.ZERO_BYTES;
-            } else {
-                InputStream is = responseEntity.getContent();
-                bytes = FileUtils.toBytes(is);
-            }
+        }   
+        try {
+            // client  can not be closed/autoclosed as it is referenced accross multiple calls by ScenarioEngine 
```

Comment:
@f-delahaye really appreciate all your comments and research above. for this comment on "client can not be autoclosed" - could you explain a bit more ?

Reply:
Basically, 
```
try (CloseableHttpClient client = clientBuilder.build()) 
```

causes `Connection pool shut down` errors.

For some reason, I thought it was not auto closed in previous version either, so I didn't really look into this, but reading the code again, I can see that it _was_ auto-closed, my bad.
I will try to take a deeper look.
Thanks!

Reply:
Ok, so calling CloseableHttpClient.close() also closes the client's connection pool.

![image](https://github.com/karatelabs/karate/assets/5797385/b16f3a33-ecd8-48e1-9523-31c224e59f9e)

So the next call to `ApacheHttpClient.invoke` will create a new `CloseableHttpClient` but will re-use the same connection pool created in `configure`, which is closed, hence the error.

Previous version of  `ApacheHttpClient` would not create the connection pool in `configure` and re-use it, instead a new one is created by every call to `invoke`. 
It probably makes sense to have the same implementation ( a new connection pool for each call) in the new version as well.


If you prefer to reuse the same `ConnectionPool`, and therefore the same `CloseableHttpClient`, one solution would be to add a close() method in `ApacheHttpClient` that would call `CloseableHttpClient.close`. But, then, the question is how/when ApacheHttpClient.close should be called...

@ptrthomas 
Any thoughts?

Reply:
first thought is have the connection pool as a thread-local - maybe this is the most efficient solution, so it never needs to be closed (will die with the JVM)

Reply:
Hummm ... interesting idea. But not sure how to implement it: whenever the CloseableHttpClient is closed, its connectionManager is automatically closed too. 

This post explains the issue better than I did:
https://stackoverflow.com/questions/76761703/correct-way-to-use-poolinghttpclientconnectionmanager-in-httpclient4-getting-iss

IMHO, `setConnectionManagerShared(true)` mentioned there is the way to go, along with a new `close` method on HttpClient that would get called from ` ScenarioEngine.stop` and would close the connection manager.
If no objection, I will commit these changes today or tomorrow for review.

Reply:
yes that sounds good

Reply:
Thanks @ptrthomas , changes have been pushed.
Note the new `HttpClient.close()` method, with no op implementation in all sub-classes except ApacheHttpClient.

As far as I am concerned, this PR is ready to be merged, if you're happy with it.

I'd still like to investigate if the connection reuse strategy may be improved. But any changes for that will come in a different PR.